app/components/post-list.js

```diff
@@ -1,4 +1,34 @@
 import Component from '@ember/component';
+import { computed, get, set } from '@ember/object';
 
 export default Component.extend({
+
+    authors: computed('posts.@each.author', {
+        get() {
+            const array = ['All']
+            this.posts.forEach(post => {
+                if(array.indexOf(post.author) === -1 && post.author) {
+                    array.push(post.author)
+                }
+                    
+            })
```

Comment:
[SUGGESTION]:
```suggestion
            return [...new Set(this.posts.map(post => post.author)];
```

Your logic works absolutely fine! This is just a simple one-liner and easy to do unique on the array. If you want to contact use the same spread operator and return :)